Formula/libvisual.rb

```diff
@@ -0,0 +1,110 @@
+class Libvisual < Formula
+  desc "Audio Visualization tool and library"
+  homepage "https://github.com/Libvisual/libvisual"
+  url "https://github.com/Libvisual/libvisual/releases/download/libvisual-0.4.2/libvisual-0.4.2.tar.gz"
+  sha256 "63085fd9835c42c9399ea6bb13a7ebd4b1547ace75c4595ce8e9759512bd998a"
+  license all_of: ["GPL-2.0-or-later", "GPL-3.0-or-later", "LGPL-2.1-or-later"]
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "xorg-server" => :test
+
+  depends_on "jack"
+  depends_on "portaudio"
+  depends_on "projectm"
+  depends_on "sdl12-compat"
+
+  uses_from_macos "bison" => :build
+
+  on_linux do
+    depends_on "bison" => :build
+    depends_on "alsa-lib"
+    depends_on "libx11"
+    depends_on "libxext"
+    depends_on "libxrender"
+    depends_on "mesa"
+    depends_on "mesa-glu"
+    depends_on "pulseaudio"
+  end
+
+  resource "libvisual-plugins" do
+    url "https://github.com/Libvisual/libvisual/releases/download/libvisual-plugins-0.4.2/libvisual-plugins-0.4.2.tar.gz"
+    sha256 "55988403682b180d0de5e6082f804f3cf066d9a08e887b10eb6a315eb40d9f87"
+  end
+
+  resource "libvisual-projectm" do
+    url "https://github.com/projectM-visualizer/frontend-libvisual-plug-in/archive/refs/tags/v2.1.1.tar.gz"
+    sha256 "eb8269c2a923546600d3f40ff90c011f03a215847215ee8bef44bfae305b4dd7"
+  end
```

Comment:
What's the release cadence of these resources relative to the main project? How do we know what versions to update these to?

Reply:
@carlocab libvisual-plugins updates with libvisual, 0.4.2 were released on the same day, will be the same for 0.4.3. For libvisual-projectm I don't expect it to move before projectM is at >=4 in here. It's independent. The reason the two are resources here is because the plugin path is owned by libvisual and the two cannot write to it during installation, at least not in Homebrew. I would love to have three formulae rather than one, but it doesn't seem possible today with the way Homebrew and Libvisual work. What do you think?

Reply:
So they need to be installed into the same prefix? If so, then this is fine. Usually for other formulae like this we just point them into some common location (e.g. `$HOMEBREW_PREFIX/share/foo`) which is where the plugin is eventually symlinked to.

Reply:
> So they need to be installed into the same prefix?

@carlocab if I understand the question right, then yes. All plugins currently need to go into the "pluginsbasedir" directory. On my local Linux shell it looks like this:

```console
# pkg-config --var pluginsbasedir libvisual-0.4
/usr/lib64/libvisual-0.4
```

> If so, then this is fine. Usually for other formulae like this we just point them into some common location (e.g. `$HOMEBREW_PREFIX/share/foo`) which is where the plugin is eventually symlinked to.

I thought about using symlinks but if the symlink target is out of the bottle scope, which package creates the symlinks and when? Something like post-inst code that even runs for bottles? Are there example formulae for this? If we can save that complexity for now, I won't be unhappy though. I'm curious about longer term options though.

Reply:
@carlocab I think I found what you are referring to:

https://github.com/Homebrew/homebrew-core/blob/0d8081340539ec8fe9094abdf5a28ec2a1773e39/Formula/vapoursynth.rb#L43

https://github.com/Homebrew/homebrew-core/blob/0d8081340539ec8fe9094abdf5a28ec2a1773e39/Formula/vapoursynth-sub.rb#L36

https://github.com/Homebrew/homebrew-core/blob/0d8081340539ec8fe9094abdf5a28ec2a1773e39/Formula/vapoursynth-ocr.rb#L39-L41

Reply:
@carlocab I tried the path with three formulae once more but I get…

> ```
> mkdir: /opt/homebrew/lib/libvisual-plugins: Operation not permitted
> ```

…[in CI](https://github.com/Homebrew/homebrew-core/actions/runs/4507647024/jobs/7935692603#step:7:525).

A user-symlink-caveat approach may be feasible for libvisual-projectm but not for libvisual-plugins: It would be too many links, 20 or so.

Should I revert back to all in one approach? Did I misuse the idea about `HOMEBREW_PREFIX`?

Also, I'm getting error `Error: libvisual-plugins is missing test dependencies: llvm mesa` while these two are installed. Does not go away with `export HOMEBREW_NO_INSTALL_FROM_API=1`. Any ideas?